### PR TITLE
Update to Aspire 8.0.2

### PR DESCRIPTION
**Pull request by joperezr**
cc: @DamianEdwards @radical

### CODE REVIEW

File: samples/AspireShop/AspireShop.ServiceDefaults/AspireShop.ServiceDefaults.csproj [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
-    <PackageReference Include="Microsoft.Extensions.ServiceDiscovery" Version="8.0.1" />
+    <PackageReference Include="Microsoft.Extensions.ServiceDiscovery" Version="8.0.2" />
     <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.9.0" />
     <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.9.0" />
     <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.9.0" />
     <PackageReference Include="OpenTelemetry.Instrumentation.GrpcNetClient" Version="1.8.0-beta.1" />
     <PackageReference Include="OpenTelemetry.Instrumentation.Http" Version="1.9.0" />
     <PackageReference Include="OpenTelemetry.Instrumentation.Runtime" Version="1.9.0" />
-    <PackageReference Include="AspNetCore.HealthChecks.Uris" Version="8.0.1" />
+    <PackageReference Include="AspNetCore.HealthChecks.Uris" Version="8.0.2" />
```
Comment (DamianEdwards) [inline samples/AspireShop/AspireShop.ServiceDefaults/AspireShop.ServiceDefaults.csproj:22]: revert this one
```suggestion
    <PackageReference Include="AspNetCore.HealthChecks.Uris" Version="8.0.1" />
```

Comment (joperezr) [reply samples/AspireShop/AspireShop.ServiceDefaults/AspireShop.ServiceDefaults.csproj:22]: Oops. missed that on my Regex 😝 Good catch.

**Comment by radical** [comment]
Can we use a property for these versions? That would be needed anyway for enabling aspire-samples CI with aspire repo.

**Comment by joperezr** [comment]
> Can we use a property for these versions? That would be needed anyway for enabling aspire-samples CI with aspire repo.

I believe the desire is for these projects to be copy-pastable (as they are samples) so by definition they don't use any sort of central package management.

**Comment by radical** [comment]
> > Can we use a property for these versions? That would be needed anyway for enabling aspire-samples CI with aspire repo.
> 
> I believe the desire is for these projects to be copy-pastable (as they are samples) so by definition they don't use any sort of central package management.

Maybe a property in each of the csproj files then, so that it would be possible cleanly replace the versions with `8.1.0-ci`(for example) on aspire CI. Or a `Directory.Build.props`?

File: samples/Metrics/ServiceDefaults/ServiceDefaults.csproj [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.6" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="8.0.6" />
     <PackageReference Include="Microsoft.Extensions.Http.Resilience" Version="8.6.0" />
-    <PackageReference Include="Microsoft.Extensions.ServiceDiscovery" Version="8.0.1" />
+    <PackageReference Include="Microsoft.Extensions.ServiceDiscovery" Version="8.0.2" />
     <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.9.0" />
     <!-- BUG: 1.8.0-rc.1 breaks prometheus exporter so sticking to 1.8.0-beta.1 for now https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1617 -->
     <PackageReference Include="OpenTelemetry.Exporter.Prometheus.AspNetCore" Version="1.8.0-rc.1" />
```
Comment (DamianEdwards) [inline samples/Metrics/ServiceDefaults/ServiceDefaults.csproj:17]: Not part of this PR but somehow this version keeps getting bumped. See the comment above about why we need it to stay the beta for now.

**Comment by DamianEdwards** [comment]
We could introduce hierarchical `Directory.Build.props` to make it easy to set one version from the root (for cases like CI) but otherwise have each sample have a single place where the version is set. That's fine as long as each sample can be copied out and run without the rest of the repo.

**Comment by radical** [comment]
```
Failed SamplesIntegrationTests.AppHostTests.TestEndpointsReturnOk(testEndpoints: VolumeMount.AppHost (1 resources)) [5 m 31 s]
  Error Message:
   Polly.Timeout.TimeoutRejectedException : The operation didn't complete within the allowed timeout of '00:05:00'.
```
.. with lots of `Azure.RequestFailedException: The API version 2024-05-04 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error. `

It's a big log, and I'm not sure if I'm looking at the correct cause for the failure.

**Comment by DamianEdwards** [comment]
Will have to try and reproduce the issue locally I think.

**Comment by joperezr** [comment]
Since I'll be out for a couple of weeks, feel free to modify or close this PR as you see fit.

**Comment by danmoseley** [comment]
https://github.com/dotnet/aspire-samples/pull/345/
